src/lib.rs

```diff
@@ -92,48 +95,58 @@ fn get_compact_symbol_table_impl(
     binary_data: &[u8],
     debug_data: &[u8],
     breakpad_id: &str,
-) -> Option<compact_symbol_table::CompactSymbolTable> {
+) -> Result<compact_symbol_table::CompactSymbolTable> {
     let mut reader = Cursor::new(binary_data);
-    if let Ok(hint) = goblin::peek(&mut reader) {
-        match hint {
-            Hint::Elf(_) => {
-                return elf::get_compact_symbol_table(binary_data, breakpad_id);
-            }
-            Hint::Mach(_) => {
-                return macho::get_compact_symbol_table(binary_data, breakpad_id);
-            }
-            Hint::MachFat(_) => {
-                let multi_arch = mach::MultiArch::new(binary_data).ok()?;
-                for fat_arch in multi_arch.iter_arches().filter_map(Result::ok) {
-                    let arch_slice = fat_arch.slice(binary_data);
-                    if let Some(table) = macho::get_compact_symbol_table(arch_slice, breakpad_id) {
-                        return Some(table);
-                    }
+    let mut first_error = None;
+    match goblin::peek(&mut reader)? {
+        Hint::Elf(_) => elf::get_compact_symbol_table(binary_data, breakpad_id),
+        Hint::Mach(_) => macho::get_compact_symbol_table(binary_data, breakpad_id),
+        Hint::MachFat(_) => {
+            let multi_arch = mach::MultiArch::new(binary_data)?;
+            for fat_arch in multi_arch.iter_arches().filter_map(std::result::Result::ok) {
+                let arch_slice = fat_arch.slice(binary_data);
+                let table = macho::get_compact_symbol_table(arch_slice, breakpad_id);
+                if table.is_err() {
+                    first_error = table.err();
+                } else {
+                    return table;
                 }
             }
-            Hint::PE => {
-                return pdb::get_compact_symbol_table(debug_data, breakpad_id);
+            if first_error.is_some() {
+                Err(first_error.unwrap())
```

Comment:
Please find a way to avoid this unwrap. You could use `unwrap_or_else` and replace the entire if condition with:

```rust
Err(first_error.unwrap_or_else(|| GetSymbolsError::InvalidInputError(...)))
```